Ensure that necessary folder exists

ClementJ35 opened the pull request:
## 🗣 Description ##

Ensure that folder exists as described in #35

ClementJ35 committed:
Ensure that necessary folder actually exists

jsf9k commented on tasks/main.yml:
```diff
@@ -21,6 +21,18 @@
+  ansible.builtin.file:
+    path: /var/log/clamav
+    state: directory
+    mode: '0755'
```
Remove unnecessary quotes:
```suggestion
    mode: 0755
```

jsf9k commented on tasks/main.yml:
```diff
@@ -21,6 +21,18 @@
+  ansible.builtin.file:
+    path: /var/log/clamav/lastscan.log
+    state: file
+    mode: '0644'
```
Remove unnecessary quotes:
```suggestion
    mode: 0644
```

jsf9k commented on tasks/main.yml:
```diff
@@ -21,6 +21,18 @@
     dest: /etc/cron.weekly/virus_scan
     mode: 0755
 
+- name: Ensure that log folder exist
```
Fix typo:
```suggestion
- name: Ensure that log folder exists
```

jsf9k reviewed (changes requested):
Requesting a few very minor changes.  Also, please address the failing tests.

ClementJ35 committed:
Change state from file to touch

"Touch" will create an empty file if the specified path does not
exist, while "file" will not.

ClementJ35 commented:
As stated here : https://docs.ansible.com/ansible/latest/collections/ansible/builtin/file_module.html, the only way to create an empty file that doesn't exist is touch. Using copy or template would result in blanking the log if it exists.
But touch seems to break idempotence autobuild test.
What next ?

ClementJ35 committed:
Remove unnecessary quotes

Co-authored-by: Shane Frasier <maverick@maverickdolphin.com>

ClementJ35 committed:
Fix typo

Co-authored-by: Shane Frasier <maverick@maverickdolphin.com>

jsf9k commented:
> As stated here : https://docs.ansible.com/ansible/latest/collections/ansible/builtin/file_module.html, the only way to create an empty file that doesn't exist is touch. Using copy or template would result in blanking the log if it exists. But touch seems to break idempotence autobuild test. What next ?

You can use [`ansible.builtin.stat`](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/stat_module.html) to determine whether you need to actually do the `touch` or not.  Let me know if you can't get it to work and I will make the change for you.

jsf9k commented:
@ClementJ35, thanks for the improvements to this Ansible role!

In the future, when accepting suggestions (like you did in commit 99749ef, for example) please change the default commit message to something more specific to the modifications in the commit.  For example, "Remove unnecessary quotes" would have been a good commit message for 99749ef.

Thanks again for the contribution...we appreciate it!

mcdonnnj commented:
@ClementJ35 Just a thought but can we use `access_time: preserve` and `modification_time: preserve` to clear idempotence while using `state: touch`?

jsf9k commented:
> @ClementJ35 Just a thought but can we use `access_time: preserve` and `modification_time: preserve` to clear idempotence while using `state: touch`?

This is worth a try, @ClementJ35.  This is a simpler and cleaner solution than what I suggested...if it works.

jsf9k committed:
Preserve access and modification times

This should make the ansible.builtin.file task idempotent.

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k commented:
@ClementJ35 - I'm sorry for my name being attached to commits 5f9ba14 and 57a0a2c.  I rebased and force-pushed in order to make minor changes to the commit messages to better align with our processes, and `git` did that.  It wasn't my intent.

Thanks for another great pull request on this repo!

mcdonnnj reviewed (approved):
Seems reasonable enough and LGTM ✔

ClementJ35 commented:
Hi @jsf9k @mcdonnnj,
Those changes are fine to me. I just didn't thought about time preservation for idempotence.
@jsf9k, don't worry about my name, I don't care. I just want that role to do better.
Thanks for your responsiveness.

dav3r reviewed (approved):
Love it!  Nice improvement.  💎
